test/packaging/windows/msi.tests.ps1

```diff
@@ -0,0 +1,128 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+function Test-Elevated {
+    [CmdletBinding()]
+    [OutputType([bool])]
+    Param()
+
+    # if the current Powershell session was called with administrator privileges,
+    # the Administrator Group's well-known SID will show up in the Groups for the current identity.
+    # Note that the SID won't show up unless the process is elevated.
+    return (([Security.Principal.WindowsIdentity]::GetCurrent()).Groups -contains "S-1-5-32-544")
+}
+
+function Invoke-Msiexec {
+    param(
+        [Parameter(ParameterSetName = 'Install', Mandatory)]
+        [Switch]$Install,
+
+        [Parameter(ParameterSetName = 'Uninstall', Mandatory)]
+        [Switch]$Uninstall,
+
+        [Parameter(Mandatory)]
+        [ValidateScript({Test-Path -Path $_})]
+        [String]$MsiPath,
+
+        [Parameter(ParameterSetName = 'Install')]
+        [HashTable] $Properties
+
+    )
+    $action = "$($PsCmdlet.ParameterSetName)ing"
+    if ($Install.IsPresent) {
+        $switch = '/I'
+    } else {
+        $switch = '/x'
+    }
+
+    $additionalOptions = @()
+    if ($Properties) {
+        foreach ($key in $Properties.Keys) {
+            $additionalOptions += "$key=$($Properties.$key)"
+        }
+    }
+
+    $argumentList = "$switch $MsiPath /quiet /l*vx $msiLog $additionalOptions"
+    $msiExecProcess = Start-Process msiexec.exe -Wait -ArgumentList $argumentList -NoNewWindow -PassThru
+    if ($msiExecProcess.ExitCode -ne 0) {
+        $exitCode = $msiExecProcess.ExitCode
+        throw "$action MSI failed and returned error code $exitCode."
+    }
+}
+
+Describe -Name "Windows MSI" -Fixture {
+    BeforeAll {
+        $msiX64Path = $env:PsMsiX64Path
+
+        # Get any existing powershell core in the path
+        $beforePath = @(([System.Environment]::GetEnvironmentVariable('PATH', 'MACHINE')) -split ';' |
+                Where-Object {$_ -like '*files\powershell*'})
+
+        $msiLog = Join-Path -Path $TestDrive -ChildPath 'msilog.txt'
+
+        foreach ($pathPart in $beforePath) {
+            Write-Warning "Found existing PowerShell path: $pathPart"
+        }
+
+        if (!(Test-Elevated)) {
```

Comment:
Can't we depend on the `RequireAdminitstrator` tag?

Reply:
No, I would have to build the infrastructure for that.  It depends on `Start-PSPester` and this isn't like the product tests.  At least for the MSI tests, everything will require admin.